### PR TITLE
Fix firebase refresh token

### PR DESCRIPTION
**Pull request by Matt75**
Do not make call without refresh token => if module is not configured or merchant logout from PsAccount
Refresh token date storage now work with multishop

### CODE REVIEW

**Comment by Quetzacoalt91** [comment]
How did you make compliant with multishop feature?

Note the tests did not pass, you need to run php-cs-fixer

**Comment by Matt75** [comment]
> How did you make compliant with multishop feature?

Replace SQL query on configuration table by adding a proper field and use Configuration class to ensure we retrieve date associated to the good shop.
New Configuration key is PS_PSX_FIREBASE_REFRESH_DATE

File: classes/Api/Firebase/Token.php [hunk 1]
```diff
@@ -48,30 +48,48 @@ public function refresh()
...
 
     /**
-     * Check the token validity. The token expire time is set to 3600 seconds.
+     * Check we can request an other token.
+     *
+     * @return bool
+     */
+    public function shouldRefreshToken()
+    {
+        return $this->hasRefreshToken() && $this->isExpired();
```
Comment (Quetzacoalt91) [inline classes/Api/Firebase/Token.php]: What if the token does not exist? `$this->hasRefreshToken()` is false, making the whole line return false too.

**Comment by Matt75** [comment]
Yes, it’s expected :)
We cannot do a call to request an access_token without a refresh_token even if token is expired.
So if false, refresh() is not called
Currently when you are not logged to Psx or you have logout, this field is empty in Configuration but getToken() call refresh() and make call without request data. As a result it throw an exception due to Bad Request, catch by GenericClient
This fix prevent that call
getToken() will return an empty string and will be considered as not logged like expected

File: classes/Api/Firebase/Token.php (hunk 1 above)
Comment (Matt75) [reply classes/Api/Firebase/Token.php]: Yes, it’s expected :)
We cannot do a call to request an access_token without a refresh_token even if token is expired.
So if false, refresh() is not called
Currently when you are not logged to Psx or you have logout, this field is empty in Configuration but getToken() call refresh() and make call without request data. As a result it throw an exception due to Bad Request, catch by GenericClient
This fix prevent that call
getToken() will return an empty string and will be considered as not logged like expected

**Comment by Quetzacoalt91** [comment]
Thank you @Matt75

**Pull request merged**